File: library/src/main/java/com/example/library/mapper/CategoryMapper.java
package com.example.library.mapper;

import java.time.LocalDateTime;

import com.example.library.dto.request.CategoryRequest;
import com.example.library.dto.response.CategoryResponse;
import com.example.library.dto.response.CategorySyncResponse;
import com.example.library.entity.Account;
import com.example.library.entity.Category;

public class CategoryMapper {

    public static CategorySyncResponse toSyncResponse(Category category) {
    	
    	return CategorySyncResponse.builder()
         .id(category.getId())
        .categoryCode(category.getCategoryCode())
        .categoryName(category.getCategoryName())
        .createdBy(category.getCreatedBy().getId())
        .version(category.getVersion())
        .updatedAt(category.getUpdatedAt())
        .deleteAt(category.getDeleteAt())
        .build();
    }
    

    public static CategoryResponse toCategoryResponse(Category category) {

        return CategoryResponse.builder()
                .id(category.getId())
                .categoryCode(category.getCategoryCode())
                .categoryName(category.getCategoryName())
                .categoryName(category.getCategoryName())
                .accountId(category.getCreatedBy().getId())
                .updatedAt(category.getUpdatedAt())
                .deleteAt(category.getDeleteAt())
                .build();
    }
    
    public static Category toCategory(CategoryRequest category, Account createdByAccount) {

        return Category.builder()
                .categoryCode(category.getCategoryCode())
                .categoryName(category.getCategoryName())
                .createdBy(createdByAccount)
                .version(0L)
                .updatedAt(LocalDateTime.now())
                .deleteAt(null)
                .build();
    }
}

File: MyApplication/app/src/main/java/com/example/myapplication/data/mapper/CategoryMapper.kt
package com.example.myapplication.data.mapper

import com.example.myapplication.data.local.entity.CategoryEntity
import com.example.myapplication.data.remote.dto.response.CategoryResponse
import com.example.myapplication.data.remote.dto.response.CategorySyncResponse
import com.example.myapplication.domain.model.Category

object categoryMapper {
    fun CategoryResponse.toCategory(): Category {
        return Category(id,categoryCode,categoryName)
    }
//    fun CategoryResponse.toEntiry(): CategoryEntity {
//        return CategoryEntity(id,categoryCode  ,categoryName,updatedAt,deleteAt,accountId)
//    }
    fun CategorySyncResponse.toEntity() = CategoryEntity(
        id = id,
        categoryCode = categoryCode,
        categoryName = categoryName,
        updatedAt = updatedAt,
        deleteAt = deleteAt,
        accountId = createdBy ,
        version
    )

    fun CategoryEntity.toCategory(): Category {
        return Category(id,categoryCode,categoryName)
    }

}